test/python/quantum_info/test_synthesis.py

```diff
@@ -75,21 +77,64 @@ def make_hard_thetas_oneq(smallest=1e-18, factor=3.2, steps=22, phi=0.7, lam=0.9
           [(0.2, 0.3, 0.1), (0.7, 0.15, 0.22), (0.001, 0.97, 2.2), (3.14, 2.1, 0.9)]]]
 
 
-class TestEulerAngles1Q(QiskitTestCase):
-    """Test euler_angles_1q()"""
+class CheckDecompositions(QiskitTestCase):
+    """Implements decomposition checkers."""
 
-    def check_one_qubit_euler_angles(self, operator, tolerance=1e-14):
+    def check_one_qubit_euler_angles(self, operator, basis=None, tolerance=1e-12):
         """Check euler_angles_1q works for the given unitary"""
-        with self.subTest(operator=operator):
-            target_unitary = operator.data
+        target_unitary = operator.data
+        if basis is None:
```

Comment:
It is not intentional.  It was part of merging methods, I change the default to 1e-14 in 5600733ae, just to be safe

Reply:
In 94f862a83 I'm setting it back to 1e-12. With 1e-14 two tests fail: `test_one_qubit_hard_thetas_zyz_basis` and `test_one_qubit_hard_thetas_zxz_basis`. That’s because `[U3Gate(smallest * factor**i, phi, lam) for i in range(steps)]` creates an angle that is too close (but not equal) to zero when `i=9`. Maybe we should round to zero angles that are too small. But that's a different issue.

In ead5c0ea9 I set different tolerance when basis or not (and remove it as parameter).